### ISSUE TITLE
Expose the current instance to custom imports

### ISSUE THREAD

**Issue by willemneal** [jtenner/as-pect#232]
Currently custom imports can be created with a function that takes a reference to the current instance's memory, but it would be even more useful if it got access to the fully constructed ASLoader constructed instance.  This would allow for using all of the exports and use the convenience functions that allow for constructing arbitrary objects and arrays, which is super useful for imports that want to inject data into the tests.

This could be solved in several ways.  

1. In TextContext before running the instance, set it to a global (not preferred, but works currently with testing)
2. Have TextContext.run be responsible for instatiating the module and then passing a reference to itself when constructing custom imports.
3. Have the custom imports be an exported class and constructing the imports would require returning an object where each import methods are bound to the class.  Then after the module is created pass the class instance a reference to it.

Personally I'm down for the third option, which could still be done with old fashioned non-class js.

**Comment by jtenner** [comment]
Prefered solution is to replace: https://github.com/jtenner/as-pect/blob/master/packages/cli/src/run.ts#L473

Create the as-pect imports and pass them down into the configuration like this:

```ts
const instance = configurationImports(runner.createImports({}), loader, wasmBuffer);
```

Then pass the instance into the `runner`.

**Comment by jtenner** [comment]
this is a breaking change and it will be included in the `4.0.0` release when the time comes.

**Comment by willemneal** [comment]
Why does it have to be a breaking change?

Could I submit a PR that tries not to be?
